src/nvim/drawline.c

```diff
@@ -1155,13 +1229,17 @@ int win_line(win_T *wp, linenr_T lnum, int startrow, int endrow, bool nochange,
                                 startrow, filler_lines, filler_todo,
                                 &c_extra, &c_final, extra, sizeof(extra),
                                 &p_extra, &n_extra, &char_attr, sign_idx,
-                                sign_cul_attr);
+                                sign_cul_attr, &statuscol);
           sign_idx++;
           if (sign_idx < wp->w_scwidth) {
             draw_state = WL_SIGN - 1;
           } else {
             sign_idx = 0;
           }
+          if (statuscol.draw) {
+            n_extra = 0;
+            continue;
+          }
```

Comment:
This block is a bit hacky:
- avoid using `continue`. Wouldn't `draw_state = WL_SIGN - 1;` work?
- `get_sign_display_info` is setting `n_extra` and then you reset it here. This kind of data flow has contributed to the complexity of this function.

Reply:
> This block is a bit hacky:

I agree, I mentioned this [here](https://github.com/neovim/neovim/pull/20621#issuecomment-1315262341).

> avoid using continue. Wouldn't draw_state = WL_SIGN - 1; work?

That doesn't seem to be sufficient no. I don't quite remember what we are skipping over by `continue`'ing. I can investigate again and see if is feasible to avoid `continue`.